tensorboard/backend/event_processing/plugin_event_accumulator.py

```diff
@@ -156,6 +156,13 @@ def __init__(
         # first event encountered per tag, so we must store that first instance of
         # content for each tag.
         self._plugin_to_tag_to_content = collections.defaultdict(dict)
+        # Locks the dict `_plugin_to_tag_to_content` (but not its individual
+        # values), as well as the `_plugin_tag_locks` dict.
```

Comment:
Is it really worth even keeping `_plugin_tag_locks` rather than just switching entirely to the coarse-grained lock?

The only thing we would lose is the ability to do the dict snapshotting of `plugin_to_tag_to_content[plugin_name]` for distinct plugins in a concurrent fashion, but given that this operation doesn't do any I/O I'm struggling to convince myself that concurrency is any benefit here, instead of just forcing those snapshots to be entirely serialized.  One can construct hypotheticals in which we have e.g. a "high-priority" plugin that might want to interrupt a "low-priority" plugin's expensive dict copy but that seems far over the line into premature optimization territory.

Reply:
Sure, that’s more than fine with me if we’re happy to do that. Any time
I _don’t_ have to think about the fact that CPython and C extensions can
throw an exception to any thread between arbitrary bytecode statements,
I’m happier for it.